synapse/lib/cell.py

```diff
@@ -2642,42 +2632,32 @@ async def iterBackupArchive(self, name, user):
 
         ctx = multiprocessing.get_context('spawn')
 
-        proc = None
-        mesg = 'Streaming complete'
-
         def getproc():
             proc = ctx.Process(target=_iterBackupProc, args=(path, linkinfo))
             proc.start()
             return proc
 
+        proc = await s_coro.executor(getproc)
         try:
-            proc = await s_coro.executor(getproc)
-
             await s_coro.executor(proc.join)
+            self.backlastuploaddt = datetime.datetime.now()
+            logger.debug(f'Backup streaming completed successfully for {name}')
 
-        except (asyncio.CancelledError, Exception) as e:
-
-            # We want to log all exceptions here, an asyncio.CancelledError
-            # could be the result of a remote link terminating due to the
-            # backup stream being completed, prior to this function
-            # finishing.
+        except Exception:
```

Comment:
I think we still need to catch CancelledErrors here so we can kill the process, currently if the remote link terminates, the process will stick around. We also don't really get much of an indication that the backup streaming failed in that case, just that it completed and the error in `_doIterBackup`:
```2024-12-19 10:26:58,157 [INFO] t2call task iterNewBackupArchive cancelled [daemon.py:t2call:MainThread:MainProcess]
2024-12-19 10:26:58,157 [ERROR] error during task _doIterBackup [Errno 104] Connection reset by peer [daemon.py:t2call:MainThread:SpawnProcess-3]
Traceback (most recent call last):
  File "/home/cisphyx/git/synapse/synapse/daemon.py", line 144, in t2call
    await link.tx(('t2:yield', {'retn': (True, item)}))
  File "/home/cisphyx/git/synapse/synapse/lib/link.py", line 166, in fini
    await self.writer.wait_closed()
  File "/home/cisphyx/.pyenv/versions/3.11.10/lib/python3.11/asyncio/streams.py", line 364, in wait_closed
    await self._protocol._get_close_waiter(self)
  File "/home/cisphyx/.pyenv/versions/3.11.10/lib/python3.11/asyncio/selector_events.py", line 999, in _read_ready__data_received
    data = self._sock.recv(self.max_size)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ConnectionResetError: [Errno 104] Connection reset by peer
2024-12-19 10:26:58,157 [INFO] Backup streaming for [/home/cisphyx/backups/cortex-4ff2353f267a3e658fce9bcbee02b4e4-1734621936034] completed. [cell.py:_iterBackupWork:MainThread:SpawnProcess-3]
```

Reply:
@Cisphyx thanks for this, I was able to reproduce it and pushed a new commit. Let me know if it seems better now.

Reply:
Looks good to me now